docs/website/contents/for-developers/NodeTasks.md

```diff
@@ -0,0 +1,303 @@
+# Overview of the tasks of a caught-up node
+
+This document gives an overview of the tasks of a caught-up node, both as a relay and as a block producer.
+
+## In a single node
+
+We refer to the [component diagram](./ComponentDiagram.md) for a very high-level overview of the data flow in a node.
+Another good diagram is [Figure 1.1. in the Shelley networking spec](https://ouroboros-network.cardano.intersectmbo.org/pdfs/network-spec/network-spec.pdf).
+
+Also see the [Glossary](./Glossary.md) for jargon explanations.
+
+### Chain selection and storage layer
+
+As a basis for all other operations, the node needs to maintain a *current selection*, ie the best valid chain it has seen so far.
+
+The selection is made up of an immutable and a volatile part:
+
+ - The immutable part of the selection consists of blocks that are no longer subject to rollback.
```

Comment:
maybe mention that this is because of the common chain prefix property in the Ouroboros papers?